### PR TITLE
runtime merge of flists

### PR DESCRIPTION
**Pull request by muhamadazmy**
- Support starting the filesystem with multiple flists to be merged in runtime
- Support layring even more flists during runtime by putting all the db paths int the `<backend>/.layered` file then signaling g8ufs with `SIGHUP`

### CODE REVIEW

File: cmd/mount.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		Cache:     cmd.Cache,
+		Target:    target,
+		Storage:   dataStore,
+		Reset:     cmd.Reset,
+	})
+}
+
+func reload(fs *g8ufs.G8ufs, cmd *Cmd) error {
+	log.Info("reload flists")
+	//load extra fslist from external file /backend/.layered
```
Comment (zaibon) [inline cmd/mount.go]: typo `fslist` -> `flist`

File: cmd/stores.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+func getMetaStore(dbs []string) (meta.MetaStore, error) {
+	var stores []meta.MetaStore
+
+	for i, db := range dbs {
+		if len(db) == 0 {
+			continue //ignore empty lines in file
+		}
+		f, err := os.Open(db)
+		if err != nil {
+			return nil, err
```
Comment (zaibon) [inline cmd/stores.go:49]: you can return without closing the file here

File: cmd/stores.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+		}
+		info, err := f.Stat()
+		if err != nil {
+			return nil, err
```
Comment (zaibon) [inline cmd/stores.go]: same here

File: cmd/stores.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+		}
+		if !info.IsDir() {
+			db += ".d"
+			dbs[i] = db //update the entry in the list
+			if err := unpack(f, db); err != nil {
+				return nil, err
```
Comment (zaibon) [inline cmd/stores.go]: and again here

File: rofs/rofs.go [hunk 5]
```diff
@@ -20,19 +20,45 @@ var (
...
+// Configuration objects can be used to manipulate some filesystem flags in runtime
+type Config struct {
 	store   meta.MetaStore
 	storage storage.Storage
+	cache   string
+}
+
+//SetMetaStore sets the filesystem meta store in runtime.
+func (c *Config) SetMetaStore(store meta.MetaStore) {
+	//TODO: should this be done atomically in a way that is synched ?
```
Comment (zaibon) [inline rofs/rofs.go:33]: I guess, from where do you read this attribute ?

**Review by zaibon — CHANGES_REQUESTED** [review]

File: rofs/rofs.go (hunk 5 above)
Comment (muhamadazmy) [reply rofs/rofs.go:33]: Well, to be honest i don't think we need to change how it's done atm. Code wise it should be safe. You can say that this todo is just a place mark if we have a problem in the future

**Review by zaibon — APPROVED** [review]

**Pull request merged**